eth2/proto_array_fork_choice/src/proto_array_fork_choice.rs

```diff
@@ -0,0 +1,699 @@
+            // We ignore the vote if it is not known in `indices`. We assume that it is outside
+            // of our tree (i.e., pre-finalization) and therefore not interesting.
+            if let Some(current_delta_index) = indices.get(&vote.current_root).copied() {
+                let delta = deltas
+                    .get_mut(current_delta_index)
+                    .ok_or_else(|| Error::InvalidNodeDelta(current_delta_index))?
+                    .checked_sub(old_balance as i64)
+                    .ok_or_else(|| Error::DeltaOverflow(current_delta_index))?;
+
+                // Array access safe due to check on previous line.
+                deltas[current_delta_index] = delta;
+            }
+
+            // We ignore the vote if it is not known in `indices`. We assume that it is outside
+            // of our tree (i.e., pre-finalization) and therefore not interesting.
+            if let Some(next_delta_index) = indices.get(&vote.next_root).copied() {
+                let delta = deltas
+                    .get(next_delta_index)
+                    .ok_or_else(|| Error::InvalidNodeDelta(next_delta_index))?
+                    .checked_add(new_balance as i64)
+                    .ok_or_else(|| Error::DeltaOverflow(next_delta_index))?;
+
+                // Array access safe due to check on previous line.
+                deltas[next_delta_index] = delta;
+            }
+
+            vote.current_root = vote.next_root;
+        }
+    }
+
+    Ok(deltas)
+}
+
+#[cfg(test)]
+mod test_compute_deltas {
+    use super::*;
+
+    /// Gives a hash that is not the zero hash (unless i is `usize::max_value)`.
+    fn hash_from_index(i: usize) -> Hash256 {
+        Hash256::from_low_u64_be(i as u64 + 1)
+    }
+
+    #[test]
+    fn zero_hash() {
+        let validator_count: usize = 16;
+
+        let mut indices = HashMap::new();
+        let mut votes = ElasticList::default();
+        let mut old_balances = vec![];
+        let mut new_balances = vec![];
+
+        for i in 0..validator_count {
+            indices.insert(hash_from_index(i), i);
+            votes.0.push(VoteTracker {
+                current_root: Hash256::zero(),
+                next_root: Hash256::zero(),
+                next_epoch: Epoch::new(0),
+            });
+            old_balances.push(0);
+            new_balances.push(0);
+        }
+
+        let deltas = compute_deltas(&indices, &mut votes, &old_balances, &new_balances)
+            .expect("should compute deltas");
+
+        assert_eq!(
+            deltas.len(),
+            validator_count,
+            "deltas should have expected length"
+        );
+        assert_eq!(
+            deltas,
+            vec![0; validator_count],
+            "deltas should all be zero"
+        );
+
+        for vote in votes.0 {
+            assert_eq!(
+                vote.current_root, vote.next_root,
+                "the vote shoulds should have been updated"
```

Comment:
```suggestion
                "the vote shouldn't have been updated"
```

Reply:
Thanks! I won't add this in because I also found this yesterday and it might make a conflict. I do appreciate these little "nitpicks", so please keep 'em coming! :)